internal/services/compute/linux_virtual_machine_scale_set_resource.go

```diff
@@ -442,6 +442,12 @@ func resourceLinuxVirtualMachineScaleSetCreate(d *pluginsdk.ResourceData, meta i
 		props.Properties.SpotRestorePolicy = spotRestorePolicy
 	}
 
+	resilientVMCreationEnabled := d.Get("resilient_vm_creation_enabled").(bool)
+	resilientVMDeletionEnabled := d.Get("resilient_vm_deletion_enabled").(bool)
+	if resiliencyPolicy := ExpandVirtualMachineScaleSetResiliency(resilientVMCreationEnabled, resilientVMDeletionEnabled); resiliencyPolicy != nil {
+		props.Properties.ResiliencyPolicy = resiliencyPolicy
+	}
```

Comment:
Seeing that `ExpandVirtualMachineScaleSetResiliency` always returns a pointer to a struct and cannot return `nil`, we can simplify here:
```suggestion
	props.Properties.ResiliencyPolicy = ExpandVirtualMachineScaleSetResiliency(d.Get("resilient_vm_creation_enabled").(bool), d.Get("resilient_vm_deletion_enabled").(bool))
```

Reply:
Fixed.